pubsub/rocketmq/metadata.go

```diff
@@ -14,68 +14,158 @@ limitations under the License.
 package rocketmq
 
 import (
-	"errors"
 	"fmt"
+	"strconv"
 
 	"github.com/dapr/components-contrib/metadata"
 	"github.com/dapr/components-contrib/pubsub"
-	"github.com/dapr/kit/logger"
-)
-
-var (
-	ErrRocketmqPublishMsg         = errors.New("rocketmq publish msg error")
-	ErrRocketmqValidPublishMsgTyp = errors.New("rocketmq publish msg error, invalid msg type")
 )
 
 const (
 	metadataRocketmqTag           = "rocketmq-tag"
 	metadataRocketmqKey           = "rocketmq-key"
 	metadataRocketmqShardingKey   = "rocketmq-shardingkey"
+	metadataRocketmqQueue         = "rocketmq-queue"
 	metadataRocketmqConsumerGroup = "rocketmq-consumerGroup"
 	metadataRocketmqType          = "rocketmq-sub-type"
 	metadataRocketmqExpression    = "rocketmq-sub-expression"
 	metadataRocketmqBrokerName    = "rocketmq-broker-name"
+	metadataRocketmqQueueID       = "rocketmq-queue-id"
 )
 
+type QueueSelectorType string
+
+const (
+	HashQueueSelector       QueueSelectorType = "hash"
+	RandomQueueSelector     QueueSelectorType = "random"
+	ManualQueueSelector     QueueSelectorType = "manual"
+	RoundRobinQueueSelector QueueSelectorType = "roundRobin"
+	DaprQueueSelector       QueueSelectorType = "dapr"
+)
+
+// RocketMQ Go Client Options
 type rocketMQMetaData struct {
-	// rocketmq Credentials
```

Comment:
Please do not remove these comments here - they are very helpful for maintaining the code and metadata. As a result it's also important you don't change the order of the metadata here.

Reply:
ok, iI'll fill in the comments